metis-repository/src/main/java/eu/europeana/metis/repository/dao/RecordDao.java

```diff
@@ -0,0 +1,61 @@
+package eu.europeana.metis.repository.dao;
+
+import com.mongodb.client.MongoClient;
+import dev.morphia.Datastore;
+import dev.morphia.Morphia;
+import dev.morphia.mapping.DiscriminatorFunction;
+import dev.morphia.mapping.MapperOptions;
+import dev.morphia.mapping.NamingStrategy;
+import eu.europeana.metis.network.ExternalRequestUtil;
+import java.util.Optional;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Database Access Object for {@link Record} instances.
+ */
+public class RecordDao {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RecordDao.class);
+
+  private final Datastore datastore;
+
+  /**
+   * Constructor.
+   *
+   * @param mongo Client to the mongo database.
+   * @param databaseName The name of the database.
+   */
+  public RecordDao(MongoClient mongo, String databaseName) {
+    final MapperOptions mapperOptions = MapperOptions.builder().discriminatorKey("className")
+        .discriminator(DiscriminatorFunction.className())
+        .collectionNaming(NamingStrategy.identity()).build();
+    this.datastore = Morphia.createDatastore(mongo, databaseName, mapperOptions);
+    this.datastore.getMapper().map(Record.class);
+  }
+
+  /**
+   * Create record in the database
+   *
+   * @param record - The record to be saved in the database
+   * @return The record just saved in the database
+   */
+  public Record createRecord(Record record) {
+
+    Optional<Record> recordFound = datastore.find(Record.class).stream()
+        .filter(x -> x.getRecordId().equals(record.getRecordId())).findFirst();
+
+    if(recordFound.isPresent()) {
+      record.setId(recordFound.get().getId());
+    } else {
+      final ObjectId objectId = Optional.ofNullable(record.getId()).orElseGet(ObjectId::new);
+      record.setId(objectId);
```

Comment:
I think if you create a new object better to not create a new one: the database will take care of it.

Reply:
I'll remove these lines of code then